cpp/src/neighbors/detail/nn_descent_gnnd.hpp

```diff
@@ -0,0 +1,220 @@
+#include <cuvs/distance/distance.hpp>
+#include <cuvs/neighbors/nn_descent.hpp>
+
+#include <raft/core/device_mdarray.hpp>
+#include <raft/core/host_mdarray.hpp>
+#include <raft/core/operators.hpp>
+#include <raft/core/pinned_mdarray.hpp>
+#include <raft/core/resource/cuda_stream.hpp>
+#include <raft/core/resources.hpp>
+#include <raft/util/pow2_utils.cuh>
```

Comment:
(nit pick) Including a .cuh header in a .hpp header might work, but I dont think it is ideal practice. It means that this script has to be compiled with nvcc. And if that is the case, it might make sense to make it a `.cuh` beforehand.

Reply:
You are right! and all we're doing with with that header is use it to round up to the next multiple of 32.
Added a small inline function `roundUp32` to do this instead.

Reply:
Is there a reason for the preference of `.hpp` over `.cuh`? Yes the roundUp is a very small function, but I believe we should be reusing what we have already in raft rather than redefining a function.

Reply:
I agree with @tarang-jain here. This should be a `.cuh` file. But whether nvcc compiles the file or gcc does does not depend on the header extension being `.hpp` or `.cuh`. Header extensions are effectively completely ignored by the compiler.

Compiler choice entirely depends on the source file extension `.cpp` or `.cu`. This header, as it defines host-device functions, needs to be included in a `.cu` file and it is so there are no errors.

Reply:
Ah right! Thanks for pointing this out @divyegala . Looks like ultimately the source file where this is being included is a `.cu`, so it works!